File: jaynetwork/src/main/java/com/jaycee/jaynetwork/manager/monitor/NetworkChangeMonitor.java
package com.jaycee.jaynetwork.manager.monitor;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.jaycee.jaynetwork.manager.listener.NetworkListener;
import com.jaycee.jaynetwork.manager.utility.NetworkUtil;

public class NetworkChangeMonitor extends BroadcastReceiver {

    private static NetworkListener networkListener;

    public NetworkChangeMonitor(NetworkListener networkListener) {
        this.networkListener = networkListener;
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        networkListener.isConnected(NetworkUtil.isConnectedToNetwork(context));
        networkListener.networkType(NetworkUtil.getConnectionType(context));
        networkListener.isConnectionFast(NetworkUtil.isConnectedFast());
    }

}

File: jaynetwork/src/main/java/com/jaycee/jaynetwork/manager/monitor/NetworkChangeMonitor.kt
package com.jaycee.jaynetwork.manager.monitor

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.jaycee.jaynetwork.manager.listener.NetworkListener
import com.jaycee.jaynetwork.manager.utility.NetworkUtil.getConnectionType
import com.jaycee.jaynetwork.manager.utility.NetworkUtil.isConnectedFast
import com.jaycee.jaynetwork.manager.utility.NetworkUtil.isConnectedToNetwork

class NetworkChangeMonitor(private val networkListener: NetworkListener) : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        networkListener.isConnected(isConnectedToNetwork(context))
        networkListener.networkType(getConnectionType(context))
        networkListener.isConnectionFast(isConnectedFast)
    }

}